client/account-recovery/reset-code-validation/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+import qs from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import EmptyContent from 'components/empty-content';
+
+import {
+	updatePasswordResetUserData,
+	setResetMethod,
+	validateRequest,
+	validateRequestError,
+} from 'state/account-recovery/reset/actions';
+
+import {
+	getAccountRecoveryValidationError,
+	getAccountRecoveryValidationKey,
+} from 'state/selectors';
+
+class ResetPasswordEmailValidation extends Component {
+	getQueryString = () => {
+		const queryString = get( window, 'location.search', null );
+
+		if ( ! queryString ) {
+			return '';
+		}
+
+		return queryString.slice( 1 );
+	}
+
+	parseQueryArgs = () => ( qs.parse( this.getQueryString() ) );
+
+	componentDidMount = () => {
+		const {
+			user,
+			method,
+			key,
+		} = this.parseQueryArgs();
+
+		this.props.validateRequest( { user }, method, key );
+	}
+
+	componentWillReceiveProps = ( nextProps ) => {
+		if ( ! this.props.validationKey && nextProps.validationKey ) {
+			const {
+				user,
+				method,
+			} = this.parseQueryArgs();
+
+			// Fill in the user & method field only if the validation is succeeded.
+			// <AccountRecoveryRoot> component is using redux state to determine which AR step we're at.
+			// Thus, we need to confirm if these data are correct before filling in, otherwise users would
+			// arrive at wrong steps.
+			this.props.updatePasswordResetUserData( { user } );
+			this.props.setResetMethod( method );
+		}
+	}
+
+	getErrorMsg = ( errorIdentifier ) => {
+		const { translate } = this.props;
+
+		switch ( errorIdentifier ) {
+			case 'RestInvalidKeyError':
```

Comment:
That was a good call. Extracted them in https://github.com/Automattic/wp-calypso/pull/13450/commits/ea57a2c7aed39a87a07bb823b81ce812148cf9ef#diff-e24dc6ce1bc226394765e25c090895b1R12 along with some other good candidates.